Nos/Models/OpenGraph/SoupOpenGraphParser.swift

```diff
@@ -3,30 +3,114 @@ import SwiftSoup
+        guard let string: String = stringValue(property, from: document) else {
             return nil
         }
-        guard let heightString = openGraphProperty(.videoHeight, from: document),
-            let height = Double(heightString) else {
+        return Double(string)
+    }
+}
+
+extension SoupOpenGraphParser {
+    /// Gets the Open Graph image metadata from the given HTML document.
+    /// - Parameter document: The HTML document.
+    /// - Returns: The Open Graph image metadata, or `nil` if none is found.
+    private func imageMetadata(from document: Document) -> OpenGraphMedia? {
+        let url = imageURL(from: document)
+        let width = doubleValue(.imageWidth, from: document)
+        let height = doubleValue(.imageHeight, from: document)
+
+        return OpenGraphMedia(url: url, width: width, height: height)
+    }
+    
+    /// Gets the Open Graph type metadata from the given HTML document.
+    /// - Parameter document: The HTML document.
+    /// - Returns: The Open Graph type metadata, or `nil` if none is found
+    private func typeMetadata(from document: Document) -> OpenGraphMediaType? {
+        guard let type: String = stringValue(.type, from: document) else {
             return nil
         }
 
-        return OpenGraphMedia(type: .video, width: width, height: height)
+        if type.starts(with: "video") {
+            return .video
+        } else if type == "website" {
+            return .website
+        } else {
+            return .unknown
+        }
+    }
+
+    /// Gets the Open Graph video metadata from the given HTML document.
+    /// - Parameter document: The HTML document.
+    /// - Returns: The Open Graph video metadata, or `nil` if none is found.
+    private func videoMetadata(from document: Document) -> OpenGraphMedia? {
+        let url = videoURL(from: document)
+        let width = doubleValue(.videoWidth, from: document)
+        let height = doubleValue(.videoHeight, from: document)
+
+        return OpenGraphMedia(url: url, width: width, height: height)
     }
 }
 
 extension SoupOpenGraphParser {
-    /// Gets the Open Graph property value from the given HTML document.
-    /// - Parameters:
-    ///   - property: The Open Graph property to fetch from the HTML document.
-    ///   - document: The HTML document.
-    /// - Returns: The value of the Open Graph property, or `nil` if none is found.
-    private func openGraphProperty(_ property: OpenGraphProperty, from document: Document) -> String? {
-        try? document.select("meta[property=\(property.rawValue)]").attr("content")
+    /// Gets the Open Graph image URL from the given HTML document.
+    /// - Parameter document: The HTML document.
+    /// - Returns: The Open Graph image URL, or `nil` if none is found.
+    /// - Note: The image URL may be in a variety of properties, including `"og:image"`, `"og:image:url`, or
+    ///        `"og:image:secure_url"`.
+    func imageURL(from document: Document) -> String? {
```

Comment:
Aren't we going to want this and the following function to return `URL`? They don't seem that useful in `String` form, but maybe I am missing something. Anyway it's an easy refactor later.

Reply:
Yeah, we definitely are. I'll do it now.

Reply:
Done! 🚀